### ISSUE TITLE
Integrate Tailwind

### ISSUE THREAD

**Issue by akesson** [leptos-rs/cargo-leptos#18]
Based on these [instructions](https://github.com/gbj/leptos/discussions/125).

Basically, install [binary](https://github.com/tailwindlabs/tailwindcss/releases) and run on CSS change.

**Comment by pepperoni21** [comment]
I think it's already possible to use Tailwind with Leptos, but I thought it would be great to have a kind of ```--tailwind``` parameter to the ```cargo leptos watch``` command to automatically run tailwindcss to watch changes without having to use two terminals. If you think it would be a good feature to add I can start working on a PR.

**Comment by akesson** [comment]
Sure. I think it would be better to add it to the config: 
```toml
[package.metadata.leptos]
tailwind = true
```
So that the `LibPackage` struct ends up with a `tailwind: bool` field.

A good second step would be to add support for automatic download and installation of the tailwind binary in the `Exe` enum.

The signalling part is a bit complex, but if you want I can do that. Just give me a shout once the above is done.

**Comment by ManitVig** [comment]
I am thinking about working on this issue. can you tell me where can I check if classes have been updates in `view!` macros,  because I think tailwind build command have to be called everytime a class attribute is updated. Having tailwind watch command running would probably interfere with cargo rebuild as it does when it tried it by just running both `tailwind watch` and `cargo leptos watch` in two terminal instances. I was getting the following error everytime I tried using a class which wasn't already cached:-

`Blocking waiting for file lock on package cache`

or am I misinterpreting. 

I am pretty new to open source so please don't mind that I am not well versed with the open source way of doing things.

**Comment by ManitVig** [comment]
I have already made the neccessary changes required for adding `tailwind:bool` argument to the conifig file and updating the `Exe` enum as well as the `ExeMeta` struct in a local branch. Shall I publish it?

**Comment by akesson** [comment]
There's a rather complex signalling system in place. See https://github.com/leptos-rs/cargo-leptos/blob/main/src/readme.md but note that it is mainly for getting an approximate understanding of it.

Let's start with what you've already done and then I'll point you to the next steps. Could you do a PR to the `tailwind-support` branch? Like that, we can take it step-by-step there.

**Comment by ManitVig** [comment]
#63

**Comment by ManitVig** [comment]
@akesson Do you think after merging changes from #62, tailwind can be integrated into the `watch` process like I mention here:-

https://github.com/leptos-rs/cargo-leptos/issues/62#issuecomment-1405975986

**Comment by akesson** [comment]
Yes, you are definitely heading the right way. Better to use `tailwind build` only because `cargo-leptos` is already doing the fs watch. I think a good next step would be to integrate tailwind into the `cargo build` step with an implementation in `src/compile/tailwind.rs`. But let's finish the PR first.
